src/test/java/core/basesyntax/report/ReportFormatterImplTest.java

```diff
@@ -0,0 +1,112 @@
+        Map<Operations, Operation> operationMap = new HashMap<>();
+        operationMap.put(Operations.B, new OperationBalance());
+        operationMap.put(Operations.P, new OperationPurchase());
+        operationMap.put(Operations.R, new OperationReturn());
+        operationMap.put(Operations.S, new OperationSupply());
+        OperationStrategy operationStrategy = new OperationStrategyImpl(operationMap);
+        reportFormatter = new ReportFormatterImpl(operationStrategy);
+        super.setUp();
+    }
+
+    public void test1ReportFormatting_Ok() {
+        String actual = reportFormatter.createReport(
+                new FileReader("src/main/resources/test1_correct.csv"));
+        String expected = "fruit,quantity" + System.lineSeparator()
+                + "apple,90" + System.lineSeparator() + "banana,152";
+        assertEquals(expected, actual);
+    }
+
+    public void test2ReportFormatting_Ok() {
+        String actual = reportFormatter.createReport(
+                new FileReader("src/main/resources/test2_correct.csv"));
+        String expected = "fruit,quantity" + System.lineSeparator()
+                + "apple,90" + System.lineSeparator() + "banana,152";
+        assertEquals(expected, actual);
+    }
+
+    public void testReportFormattingInFile_Ok() {
+        DataWriter dataWriter = new FileWriter();
+        dataWriter.writeToFile(reportFormatter.createReport(
+                new FileReader("src/main/resources/test2_correct.csv")),
+                "src/main/resources/ApplesAndBananas");
+        String actual;
+        try {
+            actual = Files.readString(Path.of("src/main/resources/ApplesAndBananas"));
+        } catch (IOException e) {
+            throw new RuntimeException("Can't correctly read data from file ApplesAndBananas", e);
+        }
+        String expected = "fruit,quantity" + System.lineSeparator()
+                + "apple,90" + System.lineSeparator() + "banana,152";
+        assertEquals(expected, actual);
+    }
+
+    public void test3ReportFormatting_incorrectOperation() {
+        try {
+            reportFormatter.createReport(
+                    new FileReader("src/main/resources/test3_incorrect.csv"));
+        } catch (RuntimeException e) {
+            assertEquals("Incorrect data", e.getMessage());
+        }
+    }
+
+    public void test4ReportFormatting_operationDontPass() {
+        try {
+            reportFormatter.createReport(
+                    new FileReader("src/main/resources/test4_incorrect.csv"));
+        } catch (RuntimeException e) {
+            assertEquals("Incorrect data", e.getMessage());
+        }
+    }
+
+    public void test5ReportFormatting_twoBalanced() {
+        try {
+            reportFormatter.createReport(
+                    new FileReader("src/main/resources/test5_incorrect.csv"));
+        } catch (RuntimeException e) {
+            assertEquals("Incorrect operation!!! This fruit also balanced!", e.getMessage());
+        }
```

Comment:
let's check this behaviour in a different test class for strategies
we can check correct behaviour as well

Reply:
I split tests into different class but how I can check the correct strategy behaviour, not in the whole context?